wasm/text/lexer.go

```diff
@@ -17,7 +17,7 @@ import (
 //
 // Note: Do not include the line and column number in a parsing error as that will be attached automatically. Line and
 // column are here for storing the source location, such as for use in runtime stack traces.
-type tokenParser func(tok tokenType, tokenBytes []byte, line, col uint32) error
+type tokenParser func(tok tokenType, tokenBytes []byte, line, col uint32) (tokenParser, error)
```

Comment:
this change reduced some complexity as we no longer have to use a module field to swap it.